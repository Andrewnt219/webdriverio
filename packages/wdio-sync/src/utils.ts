import type { NodePath } from '@babel/traverse'

export function makeParentAsync (path: NodePath) {
    const fn = path.getFunctionParent()
    if (!fn?.node.async) {
        // @ts-expect-error
        fn?.node.async = true
    }
}
