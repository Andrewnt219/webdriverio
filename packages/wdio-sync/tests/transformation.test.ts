import { assertCodeFromFixture } from './test.utils'

describe('@wdio/sync', () => {
    it('should transform browser commands', async () => {
        await assertCodeFromFixture('browser-commands')
    })
})
