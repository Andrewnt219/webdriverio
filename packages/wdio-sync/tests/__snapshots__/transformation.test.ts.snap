// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@wdio/sync should transform browser commands 1`] = `
"\\"use strict\\";

describe('webdriver.io page', () => {
  it('should have the right title', async () => {
    await browser.url('https://webdriver.io');
    const title = await browser.getTitle();
  });
});

"
`;
